serverless/question-service/functions/api/questions-summary.ts

```diff
@@ -0,0 +1,165 @@
+    const { createdAt, updatedAt, titleSlug, topicTags, ...info } = question;
+    questionsMap[titleSlug] = {
+      ...info,
+      titleSlug,
+      topicTags: topicTags.map((v) => v.name),
+    };
+  }
+
+  return questionsMap;
+}
+
+export async function upsertQuestionsSummary(
+  prisma: PrismaClient,
+  questionMap: Record<string, NormalisedQuestionSummaryType>
+) {
+  for (const [titleSlug, question] of Object.entries(questionMap)) {
+    const { topicTags, acRate, isDailyQuestion, ...info } = question;
+    try {
+      await prisma.questionSummary.upsert({
+        where: { titleSlug },
+        update: {
+          acRate,
+          isDailyQuestion, // handles updating QOTD
+          topicTags: {
+            connectOrCreate: topicTags.map((topic) => {
+              return {
+                where: { name: topic.toLowerCase() },
+                create: { name: topic.toLowerCase() },
+              };
+            }),
+          },
+        },
+        create: {
+          acRate,
+          isDailyQuestion, // in case new question becomes QOTD
+          topicTags: {
+            connectOrCreate: topicTags.map((topic) => {
+              return {
+                where: { name: topic.toLowerCase() },
+                create: { name: topic.toLowerCase() },
+              };
+            }),
+          },
+          ...info,
+        },
+      });
+    } catch (error) {
+      logger.error(`unable to upsert ${titleSlug} -- ${error}`);
+    }
+  }
+  logger.info("successfully updated data questions summary");
+}
+
+export async function purgeStaleQuestionSummary(
+  prisma: PrismaClient,
+  staleSummaryMap: Record<string, NormalisedQuestionSummaryType>
+) {
+  const oldTitleSlugs = Object.keys(staleSummaryMap);
+  if (oldTitleSlugs.length == 0) {
+    return;
+  }
+
+  // Delete entry
+  try {
+    await prisma.questionSummary.deleteMany({
+      where: {
+        titleSlug: {
+          in: oldTitleSlugs,
+        },
+      },
+    });
+  } catch (error) {
+    logger.error(error);
+  }
+}
+
+// ***** GraphQL-LeetCode ***** //
+
+const axiosInstance = axios.create({
+  headers: {
+    "Content-Type": "application/json",
+  },
+});
+
+export async function getLeetcodeQuestions() {
+  const questionMap: Record<string, NormalisedQuestionSummaryType> = {};
+  const { data } = await axiosInstance.post<QuestionSummaryListType>(
+    LEETCODE_GRAPHQL_ENDPOINT,
+    ALL_QUESTIONS_QUERY
+  );
+
+  if (!data) {
+    logger.error("unable to get LC questions");
+    return questionMap;
+  }
+
+  logger.info("successfully pulled questions summary from LeetCode");
+
+  for (const q of data.data.psetQuestionList.questions) {
+    if (q.paidOnly) {
+      continue;
+    }
+    questionMap[q.titleSlug] = {
+      acRate: q.acRate,
+      difficulty: q.difficulty,
+      hasSolution: q.hasSolution,
+      hasVideoSolution: q.hasVideoSolution,
+      paidOnly: q.paidOnly,
+      title: q.title,
+      titleSlug: q.titleSlug,
```

Comment:
if the attribute names and values are the same u can j destructure directly (if got some values u dont want copied over, can j destructure them out before doing this `...q` thing)

```suggestion
      ...q
```

Reply:
@aidanaden mentioned in the outdated one but yeah, I forgot I can use destructure it to exclude topicTags which I needed to "flatten", anw i've updated the code to follow your suggestion 👍